### ISSUE TITLE
Modern Web Acronyms list

### ISSUE THREAD

**Issue by addyosmani** [google/WebFundamentals#4061]
Inspiration: https://twitter.com/kosamari/status/822199478115586052

We could do a better job of summarizing what our 12,000 acronyms mean in a single place :)

Initially, thinking:

- H/2 - HTTP/2
- PRPL - The pattern
- RAIL - Response, Animation, Idle, Load 
- FLIP - First, Last, Invert, Play
- SPA - Single-Page App
- SW - ServiceWorker
- TTI - Time To Interactive
- FMP - First Meaningful Paint
- FCP - First Contentful Paint
- TTFP - Time to First Paint
- PWA - Progressive Web App
- TTFB - Time to first byte

**Comment by surma** [comment]
* PUSH (not an acronym, but often used without much context)
* IE
* FF
* LOL
* A2HS

**Comment by pedi** [comment]
* rAF
* rIC

**Comment by kosamari** [comment]
came here to add `rAF` and found it already listed ;)

**Comment by PaulKinlan** [comment]
@petele would love to see this live under something like /web/glossary and work out a way to auto link to them when mentioned on the site (a stretch goal maybe)

**Comment by petele** [comment]
Totally on board with this, will add it to the plan. Can I ask folks to please add definitions to their comments. I know most of them, but there are a few even I don't know!

**Comment by petele** [comment]
Also adding [Joe's list](https://docs.google.com/document/d/1yBDHBkaFOSRmBqL-rXBHL16h4F777KOUk7VnvcILgdw/edit?ts=5894c61c)

**Comment by petele** [comment]
OK, I've started the initial work for this in #4160. The glossary is a JSON array of terms and acronyms. It doesn't build the glossary yet, that'll be a next step.

Format for a term is:
```
{
 "term": "Add to Home Screen", // required
 "wordKind": "noun", // required for localization team
 "termType": "technical term", // required for localization team
 "acronyms": ["A2HS"], // optional
 "description": "TODO", // required to provide definition and for localization team
 "primaryLink": "/web/fundamentals/engage-and-retain/app-install-banners/" // optional, but link to key docs
}
```

Please feel free to add new terms and update any existing ones in the `glossary` branch.

**Comment by jpmedley** [comment]
Here's what a tech writing staff would do. In general:
* Official and well-known acronyms should always be used and never shortened. 
* If an acronym is intended to help developers remember a concept it should be spelled out on first use with the acronym in parentheses, and the acronym used thereafter. For example an article about FLIP would on first use say, "First, Last, Invert, Play (FLIP)" and "FLIP" after that.
* If it's just saving the writer from typing then it needs to go away completely. An example of this would be add to home screen. The abbreviation doesn't convey the concept and is completely opaque to anyone encountering it for the first time.

If I were king, here's how I'd apply these guidelines to the listed items.

H/2 - HTTP/2 - _Always use HTTP/2._
PRPL - The pattern - _Spell out on first use._
RAIL - Response, Animation, Idle, Load - _Spell out on first use._
FLIP - First, Last, Invert, Play - Spell out on first use.
SPA - Single-Page App
SW - ServiceWorker
TTI - Time To Interactive - Spell out on first use.
FMP - First Meaningful Paint - Spell out on first use.
FCP - First Contentful Paint - Spell out on first use.
TTFP - Time to First Paint - Spell out on first use.
PWA - Progressive Web App - Spell out on first use.
TTFB - Time to first byte - Spell out on first use.
PUSH (not an acronym, but often used without much context) as in HTTP/2 Push _Don't use. Always be specific about what kind of push and don't capitalize as though it were an acronym."_
IE - Internet Explorer - _I would generally avoid the abbreviation all together, though I suspect there's little harm in using it._
FF - Firefox - _Same._
A2HS - Add to Homescreen _Don't use_

**Comment by arthurevans** [comment]
Glossary is a great idea, look forward to seeing it.

+1 to everything Joe said.

**Comment by kaycebasques** [comment]
+1 to spelling out acronyms the first time they're used in a section

**Comment by petele** [comment]
This landed in #5003 - To add new terms or definition, edit [`src/data/glossary.yaml`](https://github.com/google/WebFundamentals/blob/master/src/data/glossary.yaml)
